test/shared/src/main/scala/zio/test/package.scala

```diff
@@ -226,37 +226,89 @@ package object test extends CompileVariants {
   ): ZIO[R, Nothing, TestResult] =
     check(rv1 <*> rv2 <*> rv3 <*> rv4)(reassociate(test))
 
+  /**
+   * A version of `check` that accepts five random variables.
+   */
+  def check[R, A, B, C, D, E](rv1: Gen[R, A], rv2: Gen[R, B], rv3: Gen[R, C], rv4: Gen[R, D], rv5: Gen[R, E])(
+    test: (A, B, C, D, E) => TestResult
+  ): ZIO[R, Nothing, TestResult] =
+    check(rv1 <*> rv2 <*> rv3 <*> rv4 <*> rv5)(reassociate(test))
+
+  /**
+   * A version of `check` that accepts six random variables.
+   */
+  def check[R, A, B, C, D, E, F](
+    rv1: Gen[R, A],
+    rv2: Gen[R, B],
+    rv3: Gen[R, C],
+    rv4: Gen[R, D],
+    rv5: Gen[R, E],
+    rv6: Gen[R, F]
+  )(
+    test: (A, B, C, D, E, F) => TestResult
+  ): ZIO[R, Nothing, TestResult] =
+    check(rv1 <*> rv2 <*> rv3 <*> rv4 <*> rv5 <*> rv6)(reassociate(test))
+
   /**
    * Checks the effectual test passes for "sufficient" numbers of samples from
    * the given random variable.
    */
-  def checkM[R, R1 <: R, E, A](rv: Gen[R, A])(test: A => ZIO[R1, E, TestResult]): ZIO[R1, E, TestResult] =
+  def checkM[R, R1 <: R, Env, A](rv: Gen[R, A])(test: A => ZIO[R1, Env, TestResult]): ZIO[R1, Env, TestResult] =
```

Comment:
This is confusing because `E` is the error type, not the environment type. I would suggest just leaving `E` as is, since it is very well known that it stands for the error type, and skipping to `F` for generic type parameters. Hopefully we don't have to add enough parameters to get to `R`. 😃

Reply:
Sorry I mixed up the error and the environment. I will change it back to `E` and use your suggestions for the extra argument names.

Reply:
I've changed `Env` back to `E` and skipped to `F` where there was a clash. I've kept `E` where the error is `Nothing`. Let me know if I should change those as well.

Reply:
Yes, I think it is slightly more consistent with for example the treatment of `mapN` in `UIO.scala` to skip `E` for a generic parameter since it typically has that other meaning though admittedly it is a very minor point. With that change should be good to go.

Reply:
Done. I've kept `E` in `reassociate` because it was `E` already before my PR and it also doesn't use the `ZIO` type.